k8s_references.go

```diff
@@ -156,3 +157,40 @@ func (ks kobjSlice) process() ([]interface{}, error) {
 	}
 	return objectRefs, nil
 }
+
+// ResourceRef is a full Kubernetes resource reference composed of Group, Version, Kind, Namespace and Name.
+type ResourceRef struct {
+	Group     string `json:"group,omitempty"`
+	Version   string `json:"version,omitempty"`
+	Kind      string `json:"kind,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+	Name      string `json:"name,omitempty"`
+}
+
+// KResourceRef returns ResourceRef from group, version, kind, namespace and name.
+func KResourceRef(group, version, kind, namespace, name string) ResourceRef {
```

Comment:
Using this function is as good (or bad) as `ResourceRef{group, version, kind, namespace, name}` - there's a considerable risk that someone will accidentally pass values in the wrong order.

Does it make sense to provide such a function when it doesn't add value?

Reply:
Probably not in this case to be honest, it's the same structure as KRef, but clearly worse because there's many more parameters, and those parameters are much more easily confused.

Reply:
Agree using the struct directly with field names spelled out is safer and independent of that this func is even slightly longer then you using the struct directly (when used without field names)

Reply:
Cool - I've removed the function in the latest version.